employee-rostering-frontend/src/domain/ShiftTemplateView.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import ShiftTemplate from './ShiftTemplate';
+import DomainObject from './DomainObject';
+import { objectWithout } from 'util/ImmutableCollectionOperations';
+import moment from 'moment';
+import DomainObjectView from './DomainObjectView';
+
+export function shiftTemplateToShiftTemplateView(shiftTemplate: ShiftTemplate): ShiftTemplateView {
+  return {
+    ...objectWithout(shiftTemplate, "spot", "rotationEmployee", "shiftTemplateDuration",
+      "durationBetweenRotationStartAndTemplateStart"),
+    spotId: shiftTemplate.spot.id as number,
+    rotationEmployeeId: shiftTemplate.rotationEmployee? shiftTemplate.rotationEmployee.id as number : null,
+    shiftTemplateDuration: shiftTemplate.shiftTemplateDuration.toISOString(),
+    durationBetweenRotationStartAndTemplateStart:
+      shiftTemplate.durationBetweenRotationStartAndTemplateStart.toISOString()
+  };
+}
+
+export function shiftTemplateViewToDomainObjectView(view: ShiftTemplateView): DomainObjectView<ShiftTemplate> {
+  return {
+    ...objectWithout(view, "spotId", "rotationEmployeeId", "shiftTemplateDuration",
+      "durationBetweenRotationStartAndTemplateStart"),
+    spot: view.spotId,
+    rotationEmployee: view.rotationEmployeeId,
+    shiftTemplateDuration: moment.duration(view.shiftTemplateDuration),
+    durationBetweenRotationStartAndTemplateStart: moment.duration(view.durationBetweenRotationStartAndTemplateStart)
+  };
+}
+
+export default interface ShiftTemplateView extends DomainObject {
+  spotId: number;
+  rotationEmployeeId: number | null;
+  shiftTemplateDuration: string;
```

Comment:
What's the reason for Duration -> string -> Duration conversion?

Reply:
Duration is serialized as a string when it is sent to frontend. Additionally, the backend expect duration as a string. So we need to convert Duration to a string and vice-versa when working with the backend. (I don't see a `Duration -> string -> Duration` here, but if it is somewhere, it because Moment.js durations are not immutable)

Reply:
By `Duration -> string -> Duration` I meant that `shiftTemplateToShiftTemplateView()` converts `Duration` to `string` and `shiftTemplateViewToDomainObjectView()` converts `string` to `Duration` so I wondered why not using `Duration` all the time.

The backend explanation partially answers that but can you tell me why this is not necessary with `ShiftView` where there is `startDateTime: Date` without the need to convert it to `string`?

Reply:
No, this is also necessary for `ShiftView` (see `KindaShiftView` in https://github.com/kiegroup/optaweb-employee-rostering/blob/react-spring-boot-refactor/employee-rostering-frontend/src/store/shift/operations.ts)

Reply:
Aah, thanks. It seems a bit inconsistent. For `ShiftView` the conversion is made inside the operation call, for `ShiftTemplateView` it's made by the translation methods. Do you think this is something that's worth aligning?

Reply:
It is confusing. Note `Shift` have four forms: `Shift`, `DomainObjectView<Shift>`. `ShiftVIew`, `KindaShiftView`. `ShiftTemplate` only have three forms: `ShiftTemplate, ShiftTemplateView, DomainObjectView<ShiftTemplate>`. Once TypeScript have suffix adding in type mapping, I can eliminate `ShiftView` and `ShiftTemplateView`. Might move the operations + define a new type in the rotation operations.

Reply:
Moved to new file(s).